source/Calamari.Shared/Integration/Scripting/WindowsPowerShell/PowerShellBootstrapper.cs

```diff
@@ -192,14 +193,41 @@ public string FormatCommandArguments(string bootstrapFile, string debuggingBoots
             commandArguments.Append("-NonInteractive ");
             commandArguments.Append("-ExecutionPolicy Unrestricted ");
 
-            var filetoExecute = IsDebuggingEnabled(variables)
+            var fileToExecute = IsDebuggingEnabled(variables)
                 ? debuggingBootstrapFile.EscapeSingleQuotedString()
                 : bootstrapFile.EscapeSingleQuotedString();
 
-            commandArguments.AppendFormat("-Command \"Try {{. {{. '{0}' -OctopusKey '{1}'; if ((test-path variable:global:lastexitcode)) {{ exit $LastExitCode }}}};}} catch {{ throw }}\"", filetoExecute, encryptionKey);
+            commandArguments.AppendFormat("-Command \"{0}Try {{. {{. '{1}' -OctopusKey '{2}'; if ((test-path variable:global:lastexitcode)) {{ exit $LastExitCode }}}};}} catch {{ throw }}\"", traceCommand, fileToExecute, encryptionKey);
             return commandArguments.ToString();
         }
 
+        static string GetPsDebugCommand(CalamariVariableDictionary variables)
+        {
+            //https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-psdebug?view=powershell-6
+            var traceArg = variables[SpecialVariables.Action.PowerShell.PSDebug.Trace];
+            var traceCommand = "-Trace 0";
+            if (int.TryParse(traceArg, out var traceArgAsInt))
+            {
+                Log.Warn($"{SpecialVariables.Action.PowerShell.PSDebug.Trace} is enabled,  This should only be used for debugging, and then disabled again for normal deployments.");
+                traceCommand = $"-Trace {traceArgAsInt}";
+            }
+            else if (bool.TryParse(traceArg, out var traceArgAsBool) && traceArgAsBool)
+            {
+                Log.Warn($"{SpecialVariables.Action.PowerShell.PSDebug.Trace} is enabled,  This should only be used for debugging, and then disabled again for normal deployments.");
+                traceCommand = "-Trace 2";
+            }
+            
+            var strictArg = variables[SpecialVariables.Action.PowerShell.PSDebug.Strict];
+            var strictCommand = "";
+            if (bool.TryParse(strictArg, out var strictArgAsBool) && strictArgAsBool)
+            {
+                Log.Info($"{SpecialVariables.Action.PowerShell.PSDebug.Strict} is enabled, putting PowerShell into strict mode where variables must be assigned a value before being referenced in a script. If a variable is referenced before a value is assigned, an exception will be thrown. This feature is experimental.");
```

Comment:
Is the experimental bit necessary/helpful, or is it likely stay experimental forever?
```suggestion
                Log.Info($"{SpecialVariables.Action.PowerShell.PSDebug.Strict} is enabled, putting PowerShell into strict mode where variables must be assigned a value before being referenced in a script. If a variable is referenced before a value is assigned, an exception will be thrown.");
```

Reply:
I added that to give us the equivalence of the "EAP" way out if things dont work so well.
Also a way to see if it works in the real world of issue diagnosis - we'll be able to remove it / change it a lot.